pkgs/nix-tools/darwin-rebuild.sh

```diff
@@ -10,10 +10,12 @@ showSyntax() {
   echo "               [{--switch-generation | -G} generation] [--verbose...] [-v...]" >&2
   echo "               [-Q] [{--max-jobs | -j} number] [--cores number] [--dry-run]" >&2
   echo "               [--keep-going] [-k] [--keep-failed] [-K] [--fallback] [--show-trace]" >&2
+  echo "               [--print-build-logs] [-L]" >&2
   echo "               [-I path] [--option name value] [--arg name value] [--argstr name value]" >&2
-  echo "               [--flake flake] [--no-flake] [--update-input input flake] [--impure]" >&2
-  echo "               [--recreate-lock-file] [--no-update-lock-file] [--refresh]" >&2
-  echo "               [--offline] [--substituters substituters-list] ..." >&2
+  echo "               [--no-flake | --flake flake] [--no-registries] [--commit-lock-file]" >&2
```

Comment:
```suggestion
  echo "               [--no-flake | [--flake flake] [--no-registries] [--commit-lock-file] ...]" >&2
```

If you wanted to be even more correct, all the flake related flags could go inside there

Reply:
I did so initially, but then I saw that `/etc/nix-darwin/flake.nix` (if it exists) will be used if `--flake` is not passed. That means that the flake related options do not depend on `--flake` being passed.

Reply:
```suggestion
  echo "               [--no-flake | [[--flake flake] [--no-registries] [--commit-lock-file] ...]" >&2
```

With an extra set of square brackets around `--flake` to imply that it's optional

Reply:
Please review my change, it's not your suggestion as-is but what I believe you meant.

Reply:
Yep that's good 👍